task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -442,19 +444,38 @@ def _create_container_definition(
                 privileged=task_config.privileged,
             )
 
+        # Set any optional requests
+        requests: Dict[str, Union[str, float]] = {}
+        if 'request_cpus' in task_config and task_config.request_cpus:
+            requests['cpus'] = task_config.request_cpus
+        if 'request_memory' in task_config and task_config.request_memory:
+            requests['memory'] = f"{task_config.request_memory}Mi"
+        if 'request_disk' in task_config and task_config.request_disk:
+            requests['ephemeral-storage'] = f"{task_config.request_disk}Mi"
```

Comment:
if we set a default value of `None` for these, we should be able to avoid the `if FOO in task_config` bits here 


it's also maybe worth testing to see what happens if you feed k8s a null request as maybe we can build the `request` dict in the same way as the `limits` one - e.g.,
```python
requests = {
            "cpu": task_config.request_cpus if task_config.request_cpu else None,
            "memory": f"{task_config.request_memory}Mi"  if task_config.request_memory else None,
            "ephemeral-storage": f"{task_config.request_disk}Mi" if task_config.request_disk else None,
}
limits = {
            "cpu": task_config.cpus,
            "memory": f"{task_config.memory}Mi",
            "ephemeral-storage": f"{task_config.disk}Mi",
}
resources = V1ResourceRequirements(
                limits=limits,
                requests=requests,
)
```

Reply:
I did try to set the default to None, iirc it caused the type system to get mad and make me do Union(None | float) kind of thing, hence this evasion.

Reply:
yea, I think without the `type=REAL_TYPE, type(None)` bit pyrsistent gets quite upset with None defaults :p

Reply:
I set this up and it works, main wrinkle is without the factory it gets upset when it gets a integer value (ie 768) for the fields. I'm not sure how much we care about this or should i go ahead and create a float_or_none factory?

Reply:
that sounds pretty reasonable! i can definitely see folks passing in integer values for these

Reply:
For the records - the testing this on jolt works. I will follow up next week with a factory for this.